query_modules/schema.cpp

```diff
@@ -108,83 +109,78 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
   record.Insert(std::string(kReturnMandatory).c_str(), mandatory);
 }
 
-struct Property {
-  std::string name;
-  mgp::Value value;
+struct PropertyInfo {
+  std::unordered_set<std::string> property_types;  // property types
+  int64_t number_of_property_occurrences = 0;
+
+  PropertyInfo() = default;
+  explicit PropertyInfo(std::string property_type)
+      : property_types({property_type}), number_of_property_occurrences(1) {}
```

Comment:
Not a mandatory change, but I would do following
```suggestion
      : property_types({std::move(property_type)}), number_of_property_occurrences(kInitialNumberOfPropertyOccurances) {}
```